proto/src/oauth2.rs

```diff
@@ -77,6 +79,9 @@ pub enum AuthorisationResponse {
     Permitted,
 }
 
+#[serde_as]
+#[skip_serializing_none]
+// this is the equivalent of serde(skip_serializing_if = "Option::is_none") applied to ALL the options
```

Comment:
That's actually cool.

Reply:
Yeah I thought about updating all our existing `serde(skip_serialize_if = "Option::is_none)`, but I think that would be better places in its own pr.